### PR TITLE
Fix issue #32466

### PR DESCRIPTION
**Pull request by briansull**
Fixes outerloop leg: R2R Windows_NT x86 Checked no_tiered_compilation @ Windows.10.Amd64.Open

Fixes issue #32466

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet/jit-contrib

File: src/coreclr/src/jit/emitxarch.cpp [hunk 1]
```diff
@@ -2835,7 +2835,13 @@ void emitter::emitHandleMemOp(GenTreeIndir* indir, instrDesc* id, insFormat fmt,
 
         // Either not generating relocatable code, or addr must be an icon handle, or the
         // constant is zero (which we won't generate a relocation for).
-        assert(!emitComp->opts.compReloc || memBase->IsIconHandle() || memBase->IsIntegralConst(0));
+        //
+        // On x86 we can have an indirection of a non-handle int constant without a reloc.
```
Comment (CarolEidt) [inline src/coreclr/src/jit/emitxarch.cpp]: Nit: With this change, it's not entirely clear that this comment is an extension of the previous comment. Also, while you're here, the previous comment isn't clear about whether this is a statement (i.e. restating the condition of the assert), or describing a condition, e.g. the if statement below).
Could you reword this as a complete statement? Perhaps:
```
// If we reach here, either:
// - we are not generating relocatable code,
// - the base address is a handle for an integer constant,
// - the base address is a constant zero (which doesn't require a reloc), or
// - the base address is a constant that fits into the memory instruction (this can happen on x86).
//   This last case is captured in the FitsInAddrBase method which is used by Lowering to determine that it can
//   be contained.
```
Of course, please correct if the above is not correct.

Comment (briansull) [reply src/coreclr/src/jit/emitxarch.cpp]: OK can change the comment.

I didn't like the existing comment, It isn't clear to me why we are testing a different set of invariants here than we when originally set the contained flag.  And actually all 31 bit constants don't require a relocation, not just zero (for both x84 and x86)

Comment (briansull) [reply src/coreclr/src/jit/emitxarch.cpp]: Maybe I will use you comment and omit this part  _(which doesn't require a reloc)_

Comment (briansull) [reply src/coreclr/src/jit/emitxarch.cpp]: OK, I changed it to:

```
        // If we reach here, either:
        // - we are not generating relocatable code, (typically the non-AOT JIT case)
        // - the base address is a handle represented by an integer constant,
        // - the base address is a constant zero, or
        // - the base address is a constant that fits into the memory instruction (this can happen on x86).
        //   This last case is captured in the FitsInAddrBase method which is used by Lowering to determine that it can
        //   be contained.
        //
```

Comment (CarolEidt) [reply src/coreclr/src/jit/emitxarch.cpp]: Great, thanks!

**Review by CarolEidt — APPROVED** [review]
LGTM

**Comment by briansull** [comment]
THe last update to this assert was this:

```
Author: Bruce Forstall <brucefo@microsoft.com>
Date:   Thu Sep 8 16:42:22 2016 -0700

    Fix dotnet/coreclr#7100

    This issue is the following assert:
    ```
    Assert failure '!emitComp->opts.compReloc || memBase->IsIconHandle()'
    ```
    while ngen'ing mscorlib on desktop.

    We're trying to generate
    ```
    cmp ebx, dword ptr [0000H]
    ```
    (due to inlining). But, emitInsBinary() doesn't expect a zero address that
    is not a relocatable handle.

    Simply change the assert to allow this. The code then generates the correct
    zero base address.


    Commit migrated from https://github.com/dotnet/coreclr/commit/08f3ae19c6e07e6772ace96706012ed87e4e6f2e
```

**Review by sandreenko — APPROVED** [review]
LGTM, thank you
